#! /bin/sh -e
## 01_alpha_symbols.dpatch by Jordi Mallach <jordi@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix use of embedded assembly so that we always use symbol versioning
## DP: instead of weak symbol aliases on alpha, since the assembly used for
## DP: weak symbols is apparently non-portable.

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad a/include/alsa-symbols.h b/include/alsa-symbols.h
--- a/include/alsa-symbols.h	2006-09-29 21:12:43.000000000 +0200
+++ b/include/alsa-symbols.h	2006-09-29 21:12:01.000000000 +0200
@@ -43,7 +43,7 @@
 	__asm__ (".symver " ASM_NAME(#real) "," ASM_NAME(#name) "@@" #version)
 #endif
 
-#ifdef USE_VERSIONED_SYMBOLS
+#if defined(USE_VERSIONED_SYMBOLS) || defined(__alpha__)
 #define use_symbol_version(real, name, version) \
 		symbol_version(real, name, version)
 #define use_default_symbol_version(real, name, version) \
